python/ray/tune/trainable.py

```diff
@@ -374,6 +377,16 @@ def restore_from_object(self, obj):
         self.restore(checkpoint_path)
         shutil.rmtree(tmpdir)
 
+    def delete_checkpoint(self, checkpoint_path):
+        """Deletes local copy of checkpoint.
+
+        Args:
+            checkpoint_path (str): Path to checkpoint.
+        """
+        checkpoint_dir = os.path.dirname(checkpoint_path)
```

Comment:
The checkpoint path is not necessarily directly under checkpoint dir

Reply:
is it okay to leave a hidden marker file in the `checkpoint_directory` to identify it and then just walk up? (see most recent commit)

i tried other methods too but nothing else seemed robust against arbitrary paths (eg if users move things around before calling `tune.run(..., restore="some/random/path")`

Reply:
I think you don't need to support the case where the user moves their restore checkpoint around outside of an existing experiment dir.